### DISCUSSION TITLE
How to acheive long text lists scrolling?

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7740] [Q&A] [answered]
![Kapture 2021-08-23 at 05 32 50](https://user-images.githubusercontent.com/87828192/130369277-3ea02599-2ae5-4181-bbc7-d4ba28e42a4a.gif)

I want to implement long text lists scrolling like this.
I think I'll have 2000~3000 text items. (PIXI.BitmapText)
What's best way to do this?
Should I use DOM?

**Comment by eXponenta** [comment, marked as answer]
DOM will also lagging.
You should use a Infinite Pooled Scroll List.
Element, that use a data provider and fixed elements count, like 10. Trick is that list items loaded while list is scrolled, and list's elements swapping between top/bottom relative a scrolling direction with replacing a text inside it.

**Reply by humbleCoder1990** [reply]
![image](https://user-images.githubusercontent.com/87828192/130371400-d881527d-34f7-4661-936e-7bb48e3c33c0.png)

Thankx eXponenta for the right direction.
Then how to implement the first item and last item's clipped text like this?

**Reply by eXponenta** [reply]
Mask on list container.
